### DISCUSSION TITLE
how to upgrade existing module created by oqtane 2.3.1 to the latest oqtane 3.0.0?

### DISCUSSION THREAD

**Discussion by Loclhh** [oqtane/oqtane.framework#1822] [Q&A] [answered]
Hi,

Is there any document that guide how to upgrade existing module created by oqtane 2.3.1 to the latest oqtane 3.0.0?
I have a custom external module created by oqtane 2.3.1 and after redeclared all framework consistent with oqtane 3.0.0, I success build it with oqtane 3.0.0. However the custom module is not recognized by the latest oqtane. I make a trick that I created the new external module in oqtane 3.0.0 with has the same name with the old module the replace it with the old one. I run oqtane server success but when I active the module on the page it get the error. I check on the database and found that oqtane didn't create any table for the module I created.

Does anyone has experience in upgrade existing module to latest oqtane framework?

Thank in advanced!

**Comment by leigh-pointer** [comment]
@Loclhh 
I am still busy moving my modules to Oqtane 3 and .net6  There are a few gotchas!

First is to update all project properties to reference the .Net6 framework.

Second is to update all the packages.
This simple solution for this is to edit the project file and modify there.

Third is the update the debug.cmd file so that the module builds to the correct bin (\bin\Debug\net6.0\)

Forth is to change the bin reference in your module .nuspec file plus update that the module is now dependent on "Oqtane.Framework" version="3.0.0"

Fith is to delete the 'obj' Folder in your Module Server folder (select your server project in Visual Studio Solution Explorer. on Toolbar click 
"show all files".)

Make sure you build before run.


Hope this helps

**Comment by hishamco** [comment]
@Loclhh I did a PR yesterday that upgrade `Oqtane.Blogs` module to `3.0.0` https://github.com/oqtane/oqtane.blogs/pull/18

**Reply by leigh-pointer** [reply]
@hishamco 
You also need to you path in the debug.cmd file and the Oqtane.Blogs.nuspec file as this is still referencing the \bin\Debug\net5.0\ path

**Reply by hishamco** [reply]
I forgot this one but I will try to simplify this by introducing props file

**Comment by sbwalker** [comment, marked as answer]
@leigh-pointer provided a good list of the changes required. The Module Creator template and Theme Creator templates were updated in this PR https://github.com/oqtane/oqtane.framework/pull/1763 if you are looking for more details. It is also important to note that a module or theme created for Oqtane 2.x will install and run on Oqtane 3.x as it is backward compatible. From a development perspective you can continue to develop your modules or themes on .NET 5 however if you want to target your testing on Oqtane 3.0.0/.NET 6 then you will need to update the paths in the debug.cmd and nuspec file so that it deploys the files to the net6.0 folder.

**Reply by Loclhh** [reply]
It's awesome with all info in your replies! Thank you very much to  leigh-pointer, hishamco and specially to sbwalker! I'm very appriciated with all hard works you did for community. I'll try my best to follow your path!

Nice day all!
